src/extension/sdk/utils.ts

```diff
@@ -237,22 +239,38 @@ export class SdkUtils {
 				this.logger.info(`Found Fuchsia project that is not vanilla Flutter`);
 		}
 
-		const flutterSdkSearchPaths = [
-			workspaceConfig?.flutterSdkHome,
-			config.flutterSdkPath,
-			// TODO: These could move into processFuchsiaWorkspace and be set on the config?
-			fuchsiaRoot && path.join(fuchsiaRoot, "lib/flutter"),
-			fuchsiaRoot && path.join(fuchsiaRoot, "third_party/dart-pkg/git/flutter"),
-			firstFlutterMobileProject,
-			firstFlutterMobileProject && extractFlutterSdkPathFromPackagesFile(path.join(firstFlutterMobileProject, ".packages")),
-			firstFlutterMobileProject && path.join(firstFlutterMobileProject, ".flutter"),
-			firstFlutterMobileProject && path.join(firstFlutterMobileProject, "vendor/flutter"),
-			process.env.FLUTTER_ROOT,
-		].concat(paths).filter(notUndefined);
-
-		const flutterSdkPath = this.findFlutterSdk(flutterSdkSearchPaths);
-		// Since we just blocked on a lot of sync FS, yield.
-		await resolvedPromise;
+		let flutterSdkPath;
+		if (fs.existsSync(flutterSnapBinPath)) {
+			flutterSdkPath = flutterSnapSdkPath;
+			// If the Flutter snap is installed but not initialised, initialise it now.
+			if (!fs.existsSync(flutterSnapSdkPath + "/.git")) {
+				await window.withProgress(
+					{
+						location: ProgressLocation.Notification,
+						title: "Initializing Flutter snap",
+					},
+					async () => {
+						await util.promisify(child_process.exec)(flutterSnapBinPath);
+					});
+			}
```

Comment:
I'm thinking this only makes sense to do if we don't find a Flutter SDK using the normal path. If a user has set up Flutter without using Snap (and added to `PATH` or set `dart.flutterSdkPath`(s)) it would be strange to kick this off and then ignore those settings.

Reply:
Good point. I'll fix this when I'm back at my desk tomorrow :)